src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState_SourceGenerators.cs

```diff
@@ -120,10 +135,13 @@ AsyncLazy<bool> GetLazy(ProjectState projectState)
 
             // Out of process, call to the remote to figure this out.
             var projectId = projectState.Id;
+            var projectStateChecksums = await projectState.GetStateChecksumsAsync(cancellationToken).ConfigureAwait(false);
+            var analyzerReferences = projectStateChecksums.AnalyzerReferences.Children;
+
             var result = await client.TryInvokeAsync<IRemoteSourceGenerationService, bool>(
                 solution,
                 projectId,
-                (service, solution, cancellationToken) => service.HasGeneratorsAsync(solution, projectId, cancellationToken),
+                (service, solution, cancellationToken) => service.HasGeneratorsAsync(solution, projectId, analyzerReferences, projectState.Language, cancellationToken),
```

Comment:
new approach passes along the analyzer-reference checksums and language, as we can answer on the oop side with just that info.